hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -535,42 +536,37 @@ public SnapshotDiffCleanupService getSnapshotDiffCleanupService() {
    * @param keyPrefix DB key prefix String
    * @return endKey String, or null if no keys with such prefix is found
    */
-  private static String findEndKeyGivenPrefix(
+  private static String performOperationGivenPrefix(
       TableIterator<String, ? extends Table.KeyValue<String, ?>> keyIter,
-      String keyPrefix) throws IOException {
-
-    String endKey;
+      String keyPrefix, ThrowableFunction<Table.KeyValue<String, ?>,
+      Void, IOException> operationFunction) throws IOException {
+    String endKey = null;
     keyIter.seek(keyPrefix);
```

Comment:
I think you can just use[TableIterator#seekToFirst()](https://github.com/apache/ozone/blob/4b9b10f6206ed5c29c0c4957a651de1d24fe86f6/hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBStoreAbstractIterator.java#L117), if you wanna get all keys starts with prefix.

Reply:
Cannot do this since we don't know what the input iterator is iterating through

Reply:
IMO it is unnecessary to use seek here and just remove`keyPrefix`. It could be simply to apply operation on each entry of iteration.
e.g.
```
  private static void performOperationGivenPrefix(
      TableIterator<String, ? extends Table.KeyValue<String, ?>> keyIter,
      ThrowableFunction<Table.KeyValue<String, ?>,
      Void, IOException> operationFunction) throws IOException {
    long startTime = System.nanoTime();
    while (keyIter.hasNext()) {
      Table.KeyValue<String, ?> entry = keyIter.next();
      operationFunction.apply(entry);
    }
    // Time took for the iterator to finish (in ns)
    long timeElapsed = System.nanoTime() - startTime;
    if (timeElapsed >= DB_TABLE_ITER_LOOP_THRESHOLD_NS) {
      // Print time elapsed
      LOG.warn("Took {} ns to find endKey. Caller is {}", timeElapsed,
          new Throwable().fillInStackTrace().getStackTrace()[1]
              .getMethodName());
    }
  }
```

If you change it to suggested, change the function name to appropriate.

Reply:
done